__tests__/lockfile.js

```diff
@@ -193,3 +193,101 @@ test('Lockfile.getLockfile (sorting)', () => {
 
   expect(actual).toEqual(expected);
 });
+
+test('parse single merge conflict', () => {
+  const file = `
+a:
+  no "yes"
+
+<<<<<<< HEAD
+b:
+  foo "bar"
+=======
+c:
+  bar "foo"
+>>>>>>> branch-a
+
+d:
+  yes "no"
+`;
+
+  const {type, object} = parse(file);
+  expect(type).toEqual('merge');
+  expect(object.a.no).toEqual('yes');
+  expect(object.b.foo).toEqual('bar');
+  expect(object.c.bar).toEqual('foo');
+  expect(object.d.yes).toEqual('no');
+});
+
+test('parse multiple merge conflicts', () => {
+  const file = `
+a:
+  no "yes"
+
+<<<<<<< HEAD
+b:
+  foo "bar"
+=======
+c:
+  bar "foo"
+>>>>>>> branch-a
+
+d:
+  yes "no"
+
+<<<<<<< HEAD
+e:
+  foo "bar"
+=======
+f:
+  bar "foo"
+>>>>>>> branch-b
+`;
+
+  const {type, object} = parse(file);
+  expect(type).toEqual('merge');
+  expect(object.a.no).toEqual('yes');
+  expect(object.b.foo).toEqual('bar');
+  expect(object.c.bar).toEqual('foo');
+  expect(object.d.yes).toEqual('no');
+  expect(object.e.foo).toEqual('bar');
+  expect(object.f.bar).toEqual('foo');
+});
+
+test('parse merge conflict fail', () => {
+  const file = `
+<<<<<<< HEAD
+b:
+  foo: "bar"
+=======
```

Comment:
Ah, it might not be obvious but in this example, the format is actually wrong because yarn lockfiles don't support colons `:`, which is why it results in a conflict.